backupccl: TestScheduleChainingLifecycle is flaky

**Describe the problem**

Observed in [this CI run](https://teamcity.cockroachdb.com/viewLog.html?buildId=6583048&buildTypeId=Cockroach_BazelEssentialCi).

```console
[TestScheduleChainingLifecycle/cluster] Failed
[23:58:19][TestScheduleChainingLifecycle/cluster] === RUN   TestScheduleChainingLifecycle/cluster
    schedule_pts_chaining_test.go:86: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/schedule_pts_chaining_test.go:86
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6617/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/schedule_pts_chaining_test.go:178
        	Error:      	Not equal: 
        	            	expected: hlc.Timestamp{WallTime:1663890902552441000, Logical:0, Synthetic:false}
        	            	actual  : hlc.Timestamp{WallTime:1663890902571559000, Logical:0, Synthetic:false}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (hlc.Timestamp) {
        	            	- WallTime: (int64) 1663890902552441000,
        	            	+ WallTime: (int64) 1663890902571559000,
        	            	  Logical: (int32) 0,
        	Test:       	TestScheduleChainingLifecycle/cluster
    --- FAIL: TestScheduleChainingLifecycle/cluster (1.00s)
```

**To Reproduce**

Attempted to repro locally via `--stress`, but couldn't observe a failure in a ~10 min run.

```bash
$ ./dev test ./pkg/ccl/backupccl --filter TestScheduleChainingLifecycle/cluster --stress
```

Jira issue: CRDB-19847

--- a/pkg/ccl/backupccl/schedule_pts_chaining_test.go
+++ b/pkg/ccl/backupccl/schedule_pts_chaining_test.go
@@ -24,6 +24,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -43,7 +44,7 @@ func (th *testHelper) createSchedules(
 	}
 	backupQuery := fmt.Sprintf("%s%s", backupStmt, backupOpts)
 	schedules, err := th.createBackupSchedule(t,
-		fmt.Sprintf("CREATE SCHEDULE FOR %s RECURRING '*/5 * * * *'", backupQuery))
+		fmt.Sprintf("CREATE SCHEDULE FOR %s RECURRING '*/2 * * * *'", backupQuery))
 	require.NoError(t, err)
 
 	// We expect full & incremental schedule to be created.
@@ -94,6 +95,19 @@ func TestScheduleChainingLifecycle(t *testing.T) {
 	th, cleanup := newTestHelper(t)
 	defer cleanup()
 
+	roundedCurrentTime := th.cfg.DB.KV().Clock().PhysicalTime().Round(time.Minute * 5)
+	if roundedCurrentTime.Hour() == 0 && roundedCurrentTime.Minute() == 0 {
+		// The backup schedule in this test uses the following crontab recurrence:
+		// '*/2 * * * *'. In english, this means "run a full backup now, and then
+		// run a full backup every day at midnight, and an incremental every 2
+		// minutes. This test relies on an incremental backup running after the
+		// first full backup. But, what happens if the first full backup gets
+		// scheduled to run within 5 minutes of midnight? A second full backup may
+		// get scheduled before the expected incremental backup, breaking the
+		// invariant the test expects. For this reason, skip the test if it's
+		// running close to midnight #spooky.
+		skip.WithIssue(t, 91640, "test flakes when the machine clock is too close to midnight")
+	}
 	th.sqlDB.Exec(t, `
 CREATE DATABASE db;
 USE db;

